product/roundhouse/infrastructure.app/ApplicationConfiguraton.cs

```diff
@@ -124,7 +124,10 @@ public static void set_defaults_if_properties_are_not_set(ConfigurationPropertyH
             }
             if (string.IsNullOrEmpty(configuration_property_holder.EnvironmentNames))
             {
-                configuration_property_holder.EnvironmentNames = ApplicationParameters.default_environment_name;
+                if (!string.IsNullOrEmpty(configuration_property_holder.EnvironmentName))
+                    configuration_property_holder.EnvironmentNames = configuration_property_holder.EnvironmentName;
+                else
+                    configuration_property_holder.EnvironmentNames = ApplicationParameters.default_environment_name;

```

Comment:
No matter what, this section needs to use the new name - think this was an oversight in the original PR (that moved to env names)